node/src/components/consensus/highway_core/state.rs

```diff
@@ -503,6 +498,35 @@ impl<C: Context> State<C> {
         }
         equivocators
     }
+
+    /// Returns an error if the vote's round exponent was changed away from the cited votes' median
+    /// by more than the allowed spread.
+    fn check_round_exp_spread(&self, wvote: &WireVote<C>) -> Result<(), VoteError> {
+        let panorama = &wvote.panorama;
+        if let Observation::Correct(vh) = &panorama[wvote.creator] {
+            if wvote.round_exp >= self.vote(vh).round_exp {
```

Comment:
Wouldn't this let pass the condition where `wvote.round_exp` is higher than allowable spread?

Reply:
Yes, that's intentional: I realized that the only thing we're worried about is the attacker going too _fast_, not too slow, so there's no point checking the other side. (So far everyone agreed with that; I think I mentioned it to @fizyk20 and @AdamGagol.)

Reply:
Doesn't it mean that enough colluding validators (by stake) could force the whole network to slow down to unusable speeds?

Reply:
But they would need to be ~50% to move the median. And then, even if they are only allowed to move it by 2 steps at a time, since the round length increases exponentially they could do that anyway.

Reply:
Unless there's an upper limit on round exp, just like there's a minimum one.

Reply:
That's a good point, yes.
On the other hand, if you control 50% you can stall the network by just turning off your nodes.

Reply:
Wouldn't you need to control 2/3 to do it?